packages/zowe-explorer/src/dataset/actions.ts

```diff
@@ -1402,3 +1436,197 @@ export async function saveFile(doc: vscode.TextDocument, datasetProvider: api.IZ
+    }
+    if (!Array.isArray(clipboardContent) && clipboardContent.memberName) {
+        return pasteMember(node, datasetProvider);
+    }
+
+    await vscode.window.withProgress(
+        {
+            location: vscode.ProgressLocation.Window,
+            title: localize("paste.dataSet.InPrg", "Uploading File(s)"),
+        },
+        async function copyDsMember() {
+            for (const content of clipboardContent) {
+                if (content.memberName) {
+                    try {
+                        await ZoweExplorerApiRegister.getMvsApi(node.getProfile()).copyDataSetMember(
+                            { dsn: content.dataSetName, member: content.memberName },
+                            { dsn: node.getLabel().toString(), member: content.memberName }
+                        );
+                    } catch (err) {
+                        vscode.window.showErrorMessage(err.message);
+                        return;
+                    }
+                }
+            }
+            datasetProvider.refreshElement(node);
+            vscode.env.clipboard.writeText("");
+        }
+    );
+}
+
+/**
+ * download given dataset node
+ *
+ * @export
+ * @param {ZoweDatasetNode} node - node to be downloaded
+ */
+export async function downloadDs(node: ZoweDatasetNode) {
+    const profile = node.getProfile();
+    let lbl: string;
+    switch (true) {
+        case contextually.isFavorite(node):
+        case contextually.isSessionNotFav(node.getParent()):
+            lbl = node.label as string;
+            break;
+        case contextually.isFavoritePds(node.getParent()):
+        case contextually.isPdsNotFav(node.getParent()):
+            lbl = node.getParent().getLabel().toString() + "(" + node.getLabel().toString() + ")";
+            break;
+        default:
+            vscode.window.showErrorMessage(localize("download.invalidNode", "downloadDataset() called with invalid node."));
+            throw Error(localize("download.invalidNode", "downloadDataset() called with invalid node. "));
+    }
+    const filePath = getDocumentFilePath(lbl, node);
+    return ZoweExplorerApiRegister.getMvsApi(profile).getContents(lbl, {
+        file: filePath,
+        returnEtag: true,
+        encoding: profile.profile.encoding,
+    });
+}
+
+/**
+ * copies given sequential dataset nodes
+ *
+ * @export
+ * @param {ZoweDatasetNode[]} nodes - nodes to be copied
+ */
+export async function copySequentialDatasets(nodes: ZoweDatasetNode[]) {
+    for (const node of nodes) {
+        try {
+            await downloadDs(node);
+            const lbl = node.getLabel().toString();
+            const inputBoxOptions: vscode.InputBoxOptions = {
+                value: lbl,
+                placeHolder: localize("pasteMember.inputBox.placeHolder", "Name of Data Set"),
+                validateInput: (text) => {
+                    return dsUtils.validateDataSetName(text) && (lbl !== text) === true ? null : "Enter valid dataset name";
```

Comment:
For the case where the destination DSN is identical to the original one (`lbl === text`), should we show a different error message?
Instead of "Enter valid dataset name", I think a more specific message like "Dataset name already exists" may be helpful.

Reply:
Thanks @t1m0thyj  for reviewing this. We need to validate new Input too and I think changing the text might be confusing. I hesitated to add "Dataset name already exists" because of this. Otherwise user would expect something like "datasets cannot start with : " or "maximum 8 chars allowed" .. and so on I think user should extract the logic what is missing/wrong the operation I'm doing at the moment if it this not extremely hidden or irrevelant to current operation.

Reply:
@t1m0thyj I got your concern. It was clear. My question was ; should we start to implement such messages ? If so I think then for consistency we need to add other ones as well. For example do we have it this one on rename function right now ?